Docs-3285-DH1

dhirsch1 opened the pull request:
This pull request is for updates made to the Logging tin Mule topic in Mule 4. Please let me know if I should add anyone else to the review.

I updated the text as suggested by Rodrigo. I also updated the ARM text based on comments from Guillermo, the Debugging Projects that Use the HTTP Connector section based on comments from Ana, and the Request and Response Logging for SOAP section based on comments from Juan Desimoni. Please let me know if we should remove the SOAP section.

I also rewrote some of the text and corrected a typo in the Latency and Concurrency graphic.

dhirsch1 committed:
first commit of the logging-in-mule topic

dhirsch1 committed:
Updating the Logging in Mule topic and fixing typo in the Latency vs Concurrenty diagram
Updating the Logging in Mule topic and fixing typo in the Latency vs Concurrenty diagram

dhirsch1 committed:
Added link to the logging topic from the TOC
Added link to the logging topic from the TOC

dhirsch1 committed:
Updated the TOC to include logging-in-mule.adoc
Updated the TOC to include logging-in-mule.adoc

dhirsch1 committed:
Entered suggested edits
Entered suggested edits

dhirsch1 committed:
Tweak
Tweak

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+endif::[]
+:keywords: mule, studio, logger, logs, log, notifications, errors, debug
+
+Mule automatically logs multiple messages and specific elements in your application flows to help you debug and keep track of events. You can also include the xref:logger-component-reference.adoc[`<logger>` Element] anywhere in a flow and set it up to output any message you want.
```
"Element" -> "element" We generally refer to this as the Logger component unless they are using the XML config for this.

dhirsch1 replied:
Fixed

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+You can create a configuration file to define:
+
+ * What kinds of messages to log
+ * How to log the messages (asynchronously or synchronously)
```
so we can log asynchronously?

dhirsch1 replied:
Yes, I checked.

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+ * How to log the messages (asynchronously or synchronously)
+ * Where to log the messages, such as to the console or disk, or to an endpoint or database.
+
+Mule uses http:ç, which is a logging facade that discovers and uses a logging strategy from the classpath, such as Apache Log4j 2 or the JDK Logger. By default, Mule includes Log4j 2, which is configured with a file called `log4j2.xml`.
```
Is `http:ç` correct?

dhirsch1 replied:
No. I will fix this.

aiannucci replied:
The logging facade is `slf4j`

dhirsch1 replied:
Fixed

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+
+The Mule server has a `log4j2.xml` in its `conf` directory, which you can customize when running the server in standalone mode. This configuration file is relevant for data output to the `mule_ee.log` file.
+
+If you remove the `çlog4j2.xml` file from your application's classpath, Mule Runtime applies a default logging configuration to the application and outputs the log to `MULE_HOME/logs/<application-name>.log`.
```
Is `çlog4j2.xml` correct?

dhirsch1 replied:
No. I will fix this. I'm not sure how that character got in.

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
++
+image::logging-in-mule-view-error-log.png[View Error Log]
+
+The output from the View Error Log is in the workspace's `.metadata/.log` file, for example:
```
"in the Studio workspace's" maybe?

dhirsch1 replied:
Alejandro, can you comment on this?

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+----
+
+After you run your Mule application, the log output is in the workspace's `.mule/logs/<project>.log` file.
+For example, for project `abcd`:
```
I don't get a logs/ subdir under the .mule dir in my workspace whether I'm running my app or not:
```
sduke-ltm:.mule sduke$ pwd
/Users/sduke/AnypointStudio/studio-workspace7/.mule
sduke-ltm:.mule sduke$ ls -la
total 0
drwxr-xr-x   3 sduke  wheel   96 Feb 11 19:58 .
drwxr-xr-x  28 sduke  wheel  896 Mar 18 17:11 ..
drwxr-xr-x  21 sduke  wheel  672 Mar  8 19:17 plugins-tmp
```
The only .log files I see in my workspace are these:
```
sduke-ltm:studio-workspace7 sduke$ find ./ -name *.log
.//.metadata/.bak_0.log
.//.metadata/.log
```

dhirsch1 replied:
Alejandro, can you comment on this?

martincousido replied:
/Applications/AnypointStudio.app/Contents/Eclipse/plugins/org.mule.tooling.server.4.1.4.ee_7.3.2.201902011858/mule/logs/{appName}.log
Logs should be inside that folder.

dhirsch1 replied:
This file is for all of the Anypoint Studio logs for all Studio versions that run Mule 4? I am seeing a slightly different file on my workspace but am not seeing a log file (maybe because I haven't turned it on):

 org.mule.tooling.server.4.1.4.ee_7.3.0.201810161635

aiannucci replied:
I think this is wrong, logs are not inside the `.mule` folder, but in `logs/{appName}.log`

dukesphere commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+== Synchronous Versus Asynchronous Logging
+
+[NOTE]
+By default, Mule logs messages asynchronously in version 3.6.0 and later.
```
This is a Mule 4 doc. I don't think we need to talk about 3.6.0.

dhirsch1 replied:
OK.

dukesphere reviewed (commented):
I'll hold off on my review until an SME has reviewed it. Rodro is out of town for a few weeks. Could Iannucci review it instead maybe?

fermujica commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+
+When logging synchronously, the execution of the thread that is processing your message is interrupted to wait for the log message to be fully handled before it can continue:
+
+image::logger-synch.jpg[logger+synch,align="center"]
```
This image is broken.

dhirsch1 replied:
The image looks fine in my build. I will contact you about this.

fermujica commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+
+When logging asynchronously, the logging operation occurs in a separate thread, so the message can process before the logging completes:
+
+image::logger-asynch-alternativo.jpg[logger+asynch+alternativo,align="center"]
```
This image is broken.

dhirsch1 replied:
The image looks fine in my build. I will contact you about this.

fermujica commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,287 @@
+
+The default configuration defines all loggers, including the root logger, as asynchronous. You can override this configuration at the domain or application level. To override this configuration at the application level, add a `logConfigFile` entry to the `mule-artiface,json` file. For example.
+
+---
```
This is causing a broken style:
<img width="640" alt="Screen Shot 2019-03-25 at 4 56 09 PM" src="https://user-images.githubusercontent.com/16763154/54949965-00f2bf80-4f1f-11e9-92d8-f60519177db2.png">

If you are referencing a JSON sample, enclose it as your JSON:

```
[source,JSON,linenums]
----
{
  "minMuleVersion": "4.0.0",

   "logConfigFile": "../../../test-classes/resources/logging/custom-log4j2.xml"
}
----
```

dhirsch1 replied:
Fixed

dhirsch1 committed:
Entered some edits
Entered somem edits

dhirsch1 committed:
Merge branch 'v4.1' into DOCS-3285-DH1

dhirsch1 committed:
Added more edits to the topic
Added more edits to the topic

dhirsch1 committed:
Entered another edit
Entered another editx

fermujica committed:
Adding missing images.

We must use `.png` files when adding screenshots. These were old images, 
taken from an old repository, and were incorrectly encoded using 
`.jpeg`.

fermujica committed:
`mule-artifact,json` should be `mule-artifact.json`

fermujica committed:
Enclosing JSON sample in a JSON codeblock for better readability and avoid broken styles in the generated HTML

dhirsch1 committed:
Fixes made to graphics that were in the wrong format. MMerge branch 'v4.1' into DOCS-3285-DH1

dhirsch1 committed:
Merge pull request #328 from mulesoft/DOCS-3285-DH1-patch

Fixes broken images and syntax

dhirsch1 committed:
Committed tweaks before pulling in graphics fixes
Committed tweaks before pullihg in graphics fixes

dhirsch1 committed:
Merged updates into my branch
Merge branch 'DOCS-3285-DH1' of github.com:mulesoft/docs-mule-runtime into DOCS-3285-DH1

dhirsch1 committed:
Tried to solve merge conflict with Fer's edits
Tried to sole merge conflict with Fer's edits

aiannucci commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,284 @@
+
+The Mule server has a `log4j2.xml` file in its `conf` directory, which you can customize when running the server in standalone mode. This configuration file is relevant for data output to the `mule_ee.log` file.
+
+If you remove the `log4j2.xml` file from your application's classpath, Mule Runtime applies a default logging configuration to the application and outputs the log to `MULE_HOME/logs/<application-name>.log`.
```
We should first mention that each application that you build on Studio, for example, comes with its own log4j2.xml, which is the file this sentence is referring to. Otherwise, right now it gets confused with the Mule server own log4j2.xml mentioned in the previous sentence.

aiannucci commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,284 @@
+
+In Anypoint Studio, `log4j2.xml` appears in the `src/main/resources` path.
+
+The default configuration defines all loggers, including the root logger, as asynchronous. You can override this configuration at the domain or application level. To override this configuration at the application level, add a `logConfigFile` entry to the `mule-artiface.json` file. For example:
```
```suggestion
The default configuration defines all loggers, including the root logger, as asynchronous. You can override this configuration at the domain or application level. To override this configuration at the application level, add a `logConfigFile` entry to the `mule-artifact.json` file. For example:
```

dhirsch1 replied:
OK

aiannucci commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,284 @@
+</Configuration>
+----
+
+The above example makes the Runtime Manager agent log its state to a rolling log file in `$MULE_HOME/logs/custom_mule_agent.log`.This file rolls over on a daily interval until it reaches a 250MB size.
```
```suggestion
The above example makes the Runtime Manager agent log its state to a rolling log file in `$MULE_HOME/logs/custom_mule_agent.log`. This file rolls over on a daily interval until it reaches a 250MB size.
```

dhirsch1 replied:
OK

aiannucci commented on modules/ROOT/pages/logging-in-mule.adoc:
```diff
@@ -0,0 +1,284 @@
+
+== Debugging Projects that Use the HTTP Connector
+
+To debug projects that use the HTTP connector, you might want to make the logging more verbose than usual and track all of the behavior of both the Http-Listener and Http-Request connectors in your project. See https://docs.mulesoft.com/connectors/http/http-troubleshooting for information.
```
Should we add xref link here or it's not possible?

dhirsch1 replied:
OK

dhirsch1 committed:
First attempt to separate application log information from runtime log i Lines starting

dhirsch1 committed:
First attempt at separating application log info from runtime log info
CFirst attempt at separating application log info from runtime log info

dhirsch1 committed:
Additional changes
Additional changesx

dhirsch1 committed:
Added an intro and tweaked some wording
Added an intro and tweaked some wording

dhirsch1 committed:
Committed changes after fixing the repo
Committed changes after fixing the repo

dhirsch1 committed:
Re-committed to make sure it worked
Re-committed to make sure it worked

dhirsch1 commented:
This PR contains technical changes that were added to this later PR: https://github.com/mulesoft/docs-mule-runtime/pull/487/files/466acf1b8d358b38de6c11afe8a2857698687c48
